src/lib/atoms/Rating/Rating.tsx

```diff
@@ -0,0 +1,321 @@
+import React, {
+    FC,
+    MouseEvent,
+    ReactNode,
+    cloneElement,
+    isValidElement,
+    useLayoutEffect,
+    useState,
+    JSX,
+    KeyboardEvent
+} from 'react';
+import classNames from 'classnames';
+
+// Components
+import SvgSquareIcon from './DefaultSvg';
+
+// Styles
+import './Rating.scss';
+
+const sizes = { small: 32, medium: 36, big: 42 };
+
+export interface IRatingProps {
+    /*
+     * The default rating value is selected when the component is first rendered.
+     */
+    defaultValue?: number;
+
+    /**
+     * The currently selected rating value. Use this to control the component from the outside.<br>
+     * If this prop is used the component will lose default behavior related to rating state.
+     */
+    value?: number;
+
+    /**
+     * The number of rating elements to render.
+     */
+    count?: number;
+
+    /**
+     * The icon or character used to represent each rating element.<br>
+     * It can be a `React functional component`, `string`, `number`, or a `function` that returns a React node.<br>
+     * Possible values: `FC | string | number | ((i: number) => ReactNode)`
+     */
+    character?: FC | string | number | ((i: number) => ReactNode);
+
+    /**
+     * The color is used for the rating elements when hovered over or selected.
+     */
+    color?: string;
+
+    /**
+     * The background color of the rating elements.
+     */
+    bgColor?: string;
+
+    /**
+     * Allows users to select half values in the rating component, enabling finer granularity.<br>
+     * For example, if `count` is 5, users can select 1, 1.5, 2, 2.5, and so on up to 5.
+     */
+    halfAllow?: boolean;
+
+    /**
+     * When set to `true`, the rating component becomes read-only, preventing any interaction.
+     */
+    readonly?: boolean;
+
+    /**
+     * The size of the rating elements.<br>
+     * Possible values: `small | medium | big`
+     */
+    size?: 'small' | 'medium' | 'big';
+
+    /**
+     * Callback function that is called when the rating value changes.<br>
+     * Receives the new rating value as an argument.
+     */
+    onChange?: (rating: number) => void;
+}
+
+interface IElementProps {
+    isIcon?: boolean;
+    Element: JSX.Element | string | number;
+    compareElement?: boolean;
+    color: string;
+    bgColor: string;
+    size: string;
+}
+
+const Element: FC<IElementProps> = ({ isIcon = false, Element, compareElement = false, color, bgColor, size }) => {
+    const currentColor = compareElement ? color : bgColor;
+
+    const className = classNames('rating__icon', `s-${size}`);
```

Comment:
Why do you need to use the `classNames` function here?
```suggestion
    const className = `rating__icon s-${size}`;
```